linker error during `make check` under macOS (duplicate symbol `_d_randtest2`)

A colleague of mine is trying to check FLINT built under macOS.  He does not hit #2048, perhaps because his architecture is `x86_64` rather than `aarch64` ... ?  But he encounters a linker error at a later stage:

```
$ make check
<...>
   CC  mag/test/main.c
duplicate symbol '_d_randtest2' in:
     /private/var/folders/rr/q47s6xq93n1_lhj48t9rmqkr0000gn/T/main-4a8524.o
     ./libflint.a[85](double_interval_merged.o)
ld: 1 duplicate symbols
clang: error: linker command failed with exit code 1 (use -v to see 
invocation)
make: *** [build/mag/test/main] Error 1
```

Indeed, I see `d_randtest2` defined twice, but perhaps other linkers have ignored it.  He is, like me, using the native Apple LLVM, not Homebrew LLVM ...

--- a/src/mag/test/t-d_log_lower_bound.c
+++ b/src/mag/test/t-d_log_lower_bound.c
@@ -23,7 +23,7 @@
 #define EXP_MINUS_32 2.3283064365386962891e-10
 #define EXP_MINUS_64 5.42101086242752217e-20
 
-double
+static double
 d_randtest2(flint_rand_t state)
 {
     ulong m1, m2;

--- a/src/mag/test/t-d_log_upper_bound.c
+++ b/src/mag/test/t-d_log_upper_bound.c
@@ -23,7 +23,7 @@
 #define EXP_MINUS_32 2.3283064365386962891e-10
 #define EXP_MINUS_64 5.42101086242752217e-20
 
-double
+static double
 d_randtest2(flint_rand_t state)
 {
     ulong m1, m2;

--- a/src/mag/test/t-set_d.c
+++ b/src/mag/test/t-set_d.c
@@ -23,7 +23,7 @@
 #define EXP_MINUS_32 2.3283064365386962891e-10
 #define EXP_MINUS_64 5.42101086242752217e-20
 
-double
+static double
 d_randtest2(flint_rand_t state)
 {
     ulong m1, m2;

--- a/src/mag/test/t-set_d_2exp_fmpz.c
+++ b/src/mag/test/t-set_d_2exp_fmpz.c
@@ -23,7 +23,7 @@
 #define EXP_MINUS_32 2.3283064365386962891e-10
 #define EXP_MINUS_64 5.42101086242752217e-20
 
-double
+static double
 d_randtest2(flint_rand_t state)
 {
     ulong m1, m2;

